File: app/src/main/java/org/schabi/newpipe/settings/BasePreferenceFragment.java
package org.schabi.newpipe.settings;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;
import androidx.preference.PreferenceManager;

import org.schabi.newpipe.MainActivity;
import org.schabi.newpipe.util.ThemeHelper;

import java.util.Objects;

public abstract class BasePreferenceFragment extends PreferenceFragmentCompat {
    protected final String TAG = getClass().getSimpleName() + "@" + Integer.toHexString(hashCode());
    protected final boolean DEBUG = MainActivity.DEBUG;

    SharedPreferences defaultPreferences;

    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        defaultPreferences = PreferenceManager.getDefaultSharedPreferences(requireActivity());
        super.onCreate(savedInstanceState);
    }

    @Override
    public void onViewCreated(@NonNull final View rootView,
                              @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(rootView, savedInstanceState);
        setDivider(null);
        ThemeHelper.setTitleToAppCompatActivity(getActivity(), getPreferenceScreen().getTitle());
    }

    @Override
    public void onResume() {
        super.onResume();
        ThemeHelper.setTitleToAppCompatActivity(getActivity(), getPreferenceScreen().getTitle());
    }

    @NonNull
    public final Preference requirePreference(@StringRes final int resId) {
        final Preference preference = findPreference(getString(resId));
        Objects.requireNonNull(preference);
        return preference;
    }
}

File: app/src/main/java/org/schabi/newpipe/settings/BasePreferenceFragment.kt
package org.schabi.newpipe.settings

import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import androidx.annotation.StringRes
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import androidx.preference.PreferenceManager
import org.schabi.newpipe.MainActivity
import org.schabi.newpipe.util.ThemeHelper
import java.util.Objects

abstract class BasePreferenceFragment() : PreferenceFragmentCompat() {
    protected val TAG: String = javaClass.getSimpleName() + "@" + Integer.toHexString(hashCode())
    var defaultPreferences: SharedPreferences? = null
    public override fun onCreate(savedInstanceState: Bundle?) {
        defaultPreferences = PreferenceManager.getDefaultSharedPreferences(requireActivity())
        super.onCreate(savedInstanceState)
    }

    protected fun addPreferencesFromResourceRegistry() {
        addPreferencesFromResource(
                SettingsResourceRegistry.Companion.getInstance().getPreferencesResId(this.javaClass))
    }

    public override fun onViewCreated(rootView: View,
                                      savedInstanceState: Bundle?) {
        super.onViewCreated(rootView, savedInstanceState)
        setDivider(null)
        ThemeHelper.setTitleToAppCompatActivity(getActivity(), getPreferenceScreen().getTitle())
    }

    public override fun onResume() {
        super.onResume()
        ThemeHelper.setTitleToAppCompatActivity(getActivity(), getPreferenceScreen().getTitle())
    }

    fun requirePreference(@StringRes resId: Int): Preference {
        val preference: Preference? = findPreference(getString(resId))
        Objects.requireNonNull(preference)
        return (preference)!!
    }

    companion object {
        protected val DEBUG: Boolean = MainActivity.Companion.DEBUG
    }
}
